File: backend/src/main/java/com/gpuoj/model/Problem.java
package com.gpuoj.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.time.OffsetDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Table("problems")
public class Problem {
    @Id
    private String id;
    private String title;
    private String difficulty;
    private String judgeType;
    private int timeLimitMs;
    private Integer vramLimitMb;
    private Double speedupThreshold;
    private String gitCommitHash;
    private OffsetDateTime syncedAt;
}

File: backend/src/main/kotlin/com/gpuoj/model/Problem.kt
package com.gpuoj.model

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.time.OffsetDateTime

@Table("problems")
data class Problem(
    @Id var id: String? = null,
    var title: String? = null,
    var difficulty: String? = null,
    var judgeType: String? = null,
    var timeLimitMs: Int = 0,
    var vramLimitMb: Int? = null,
    var speedupThreshold: Double? = null,
    var gitCommitHash: String? = null,
    var syncedAt: OffsetDateTime? = null
)
